examples/cpp/core/bf/main.cpp

```diff
@@ -226,11 +231,32 @@ int main(int argc, char** argv)
         {
             InputFile input(options.file);
 
-            if (options.from == guess)
-                std::cerr << "Guessed " << ToString(options.from = Guess(input)) << std::endl;
+            do
+            {
+                // In order to decode multiple payloads from a file we need to
+                // use InputFile& however that usage doesn't support marshalled
+                // bonded<T> in untagged protocols. As a compromise we use
+                // InputFile for the last payload and InputFile& otherwise.
+                if (options.schema.size() > 1 || options.from.size() > 1)
+                {
+                    if (!Transcode<InputFile&>(input, options))
+                        return 1;
+                }
+                else
+                {
+                    if (!Transcode<InputFile>(input, options))
+                        return 1;
+                }
+
+                if (!options.schema.empty())
+                    options.schema.pop_front();
+
+                if (!options.from.empty())
+                    options.from.pop_front();
+            }
+            while (!options.schema.empty() || !options.from.empty());
```

Comment:
An extension, if you're interested, would be to have a way to specify that the last from/schema pair should be used to continually transcode until EOF.